src/Stratis.Features.FederatedPeg/FederationGatewayFeature.cs

```diff
@@ -126,6 +130,15 @@ internal class FederationGatewayFeature : FullNodeFeature
 
         public override async Task InitializeAsync()
         {
+            if (this.federationManager?.CurrentFederationKey?.PubKey != null &&
+                this.federationManager.CurrentFederationKey.PubKey.ToString() != this.federationGatewaySettings.PublicKey)
```

Comment:
I always thought that `this.federationGatewaySettings.PublicKey` can be different from the key for block creation. 
If not then what's the point storing it in 2 places at once?

Reply:
I believe the `.dat` file exists to hold the private key. It just happens that the public key can also be derived from the private key. @codingupastorm , maybe you can chime in here. All I can say is that if these public keys are different then we run into problems verifying mined blocks - i.e. that they were mined by the correct federation member (as determined by block time).

Reply:
.dat file is for PoA. federationGatewaySettings.PublicKey is used for multisig.